src/cursor/cur_file.c

```diff
@@ -59,6 +61,7 @@ WT_STAT_USECS_HIST_INCR_FUNC(opwrite, perf_hist_opwrite_latency)
         op;                                                                                 \
         if (__saved_txn != NULL) {                                                          \
             (session)->txn = __saved_txn;                                                   \
+            F_CLR((session), WT_SESSION_NO_RECONCILE);                                      \
```

Comment:
Is there a chance that `SESSION_NO_RECONCILE` was set before the `WT_WITH_CHECKPOINT` and this will clear it incorrectly?

Reply:
I don't think this can happen. This macro is only called in the outer layer of the api call. The other case that we set WT_SESSION_NO_RECONCILE is when we do reentrant reconciliation.

Reply:
That's a good point. I'll handle it.

Reply:
I think the existing code already handles the reentrant case.

```
           if (F_ISSET(__saved_txn, WT_TXN_IS_CHECKPOINT)) {                               \
                WT_ASSERT(                                                                  \
                  session, (cbt)->checkpoint_write_gen == (session)->checkpoint_write_gen); \
                __saved_txn = NULL;                                                         \
            }
```
This code will disable the clearing code being triggered by setting __saved_txn = NULL in the reentrant case. Therefore, the flag is only cleared when we exit the external api call.

Reply:
I have to admit that I'm not really following. I believe that current usages of this are safe, but it still seems pragmatic to be safe and save/restore, rather than unconditionally clear on exit. If we don't do that future users of the macro need to be aware (somehow) that the `WT_SESSION_NO_RECONCILE` flag needs to be not-set on entry.

Reply:
This makes sense. Fixed. Thank you.